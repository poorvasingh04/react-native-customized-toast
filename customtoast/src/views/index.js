import Toast from './toast/Toast';
import Theme from './toast/theme';
import { Position } from './toast/Position';
export {
  Toast,
  Theme,
  Position
};