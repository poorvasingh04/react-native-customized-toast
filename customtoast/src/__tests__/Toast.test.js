import React from 'react';
import {shallow} from 'enzyme';
import { TouchableOpacity } from 'react-native';
import { Toast } from '../views';

describe('Toast unit tests------', () => {
  var wrapper;
  var useEffect;
  var props;

  const mockUseEffect = () => {
    useEffect.mockImplementationOnce(f => f());
  };

  beforeEach(() => {
    useEffect = jest.spyOn(React, "useEffect");
    props = {
      message: { text: 'Test toast' },
    };
    mockUseEffect();
    mockUseEffect();
    wrapper = shallow(<Toast {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match to snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should hide toast', () => {
    let component = shallow(<Toast />);
    expect(component.find(TouchableOpacity).length).toBe(0);
  });

  it('should show toast', () => {
    expect(props.message).toEqual({ text: 'Test toast' });
    expect(wrapper.find(TouchableOpacity).length).toBe(1);
  });

  it('should update toast message', () => {
    const message = { text: 'Update toast' };
    wrapper.setProps({ message });
    const Text = wrapper.find('Text');
    expect(Text.length).toBe(1);
    expect(Text.props().children).toEqual('Update toast'); 
  });

  it('should hide toast on tap', () => {
    const button = wrapper.find(TouchableOpacity);
    button.props().onPress();
    expect(wrapper.find(TouchableOpacity).length).toBe(0);
  });

});
