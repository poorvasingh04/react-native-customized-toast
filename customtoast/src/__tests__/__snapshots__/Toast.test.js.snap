// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast unit tests------ should match to snapshot 1`] = `ShallowWrapper {}`;
